org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundSpgIterator.java

```diff
@@ -63,19 +62,32 @@ public CompoundSpgIterator(CompoundGenerator gen) {
 			}
 		}
 
-		JythonObjectFactory<ScanPointIterator> compoundGeneratorFactory = ScanPointGeneratorFactory.JCompoundGeneratorFactory();
+		final JythonObjectFactory<ScanPointIterator> compoundGeneratorFactory = ScanPointGeneratorFactory.JCompoundGeneratorFactory();
 
-        Object[] excluders = getExcluders(gen.getModel().getRegions());
-        Object[] mutators = getMutators(gen.getModel().getMutators());
-        double duration = gen.getModel().getDuration();
+        final Object[] excluders = getExcluders(gen.getModel().getRegions());
+        final Object[] mutators = getMutators(gen.getModel().getMutators());
+        final double duration = gen.getModel().getDuration();
+        final boolean continuous = isContinuous();
 
-        ScanPointIterator iterator = compoundGeneratorFactory.createObject(
-				iterators, excluders, mutators, duration);
+        final ScanPointIterator iterator = compoundGeneratorFactory.createObject(
+				iterators, excluders, mutators, duration, continuous);
 
-        index = -1;
         pyIterator = iterator;
 	}
 
+	/**
+	 * A Compound model is continuous if it's inner most model is a map model and that model is continuous.
+	 * @return
+	 */
+	private boolean isContinuous() {
+		List<Object> models = gen.getModel().getModels();
+		if (!models.isEmpty()) {
+			final Object innerModel = models.get(models.size() - 1);
+			return innerModel instanceof IMapPathModel && ((IMapPathModel) innerModel).isContinuous();
+		}
+		return false;
+	}
+
 	private IPosition createFirstPosition(){
 	    IPosition pos = new MapPosition();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "pos" which hides the field declared at line 50. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)